address.md

```diff
@@ -0,0 +1,303 @@
+    
+    // 0: ID
+    // 1: Blake2b160-Hash of secp256k1 Public Key
+    // 2: Blake2b160-Hash of Actor creation data
+    // 3: BLS Public Key
+    Protocol byte
+    
+    // raw bytes containing the data associated with protocol
+    Payload []byte
+}
+```
+
+#### Network Prefix
+
+The **network prefix** is prepended to an address when encoding to a string. The network prefix indicates which network an address belongs in. The network prefix may either be `f` for filecoin mainnet or `t` for filecoin testnet. Is it worth noting that a network prefix will never appear on chain and is only used when encoding an address to a human readable format.
+
+#### Protocol Indicator
+
+The **protocol indicator** byte describes how a method should interpret the information in the payload field of an address. Any deviation for the algorithms and data types specified by the protocol must be assigned a new protocol number. In this way, protocols also act as versions.
+
+- `0`: ID
+- `1` : SECP256K1 Public Key
+- `2` : Actor
+- `3` : BLS Public Key
+
+An example description in golang:
+
+```go
+// Protocol byte
+type Protocol = byte
+const (
+    ID Protocol = iota
+    SECP256K1
+    Actor
+    BLS
+)
+```
+
+###### Protocol 0: IDs
+
+**Protocol 0** addresses are simple IDs. These addresses are expected to remain within the network. All actors have a numeric ID even if they don't have public keys. The payload of an ID address is encoded as decimal numbers '0'-'9'. IDs are not hashed and do not have a checksum.
+
+**Bytes**
+
+```
+|----------|---------|
+| protocol | payload |
+|----------|---------|
+|    0     | '0'-'9' |
+```
+
+**String**
+
+```
+|------------|----------|---------|
+|  network   | protocol | payload |
+|------------|----------|---------|
+| 'f' or 't' |     0    | '0'-'9' |
+```
+
+###### Protocol 1: libsecpk1 Elliptic Curve Public Keys
+
+**Protocol 1** addresses represent secp256k1 public encryption keys. The payload field contains the [Blake2b 160](https://blake2.net/) hash of the public key. 
+
+**Bytes**
+
+```
+|----------|---------------------|
+| protocol |        payload      |
+|----------|---------------------|
+|    1     | blake2b-160(PubKey) |
+```
+
+**String**
+
+```
+|------------|----------|---------------------|----------|
+|  network   | protocol |      payload        | checksum |
+|------------|----------|---------------------|----------|
+| 'f' or 't' |     1    | blake2b-160(PubKey) |  4 bytes |
+                  base32[................................]
+```
+
+###### Protocol 2: Actor
+
+**Protocol 2** addresses representing an Actor. The payload field contains the [Blake2b 160](https://blake2.net/) hash of meaningful data produced as a result of creating the actor. 
+
+```
+|------------|----------|---------------------|----------|
+|  network   | protocol |      payload        | checksum |
+|------------|----------|---------------------|----------|
+| 'f' or 't' |     2    |  blake2b-160(TODO)  |  4 bytes |
```

Comment:
Need to address this TODO cc @whyrusleeping, what values are we hasing here?

Reply:
We arent hashing anything, its just random bytes that don't mean anything.

Reply:
(technically, we get those random bytes from hashing something, but thats an implementation detail that I think can be left out of the spec)